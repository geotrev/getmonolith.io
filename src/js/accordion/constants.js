export const Selectors = {
  // unique
  DATA_ACCORDION: "data-accordion",
  DATA_COLLAPSIBLE: "data-collapsible",
  DATA_TOGGLE_MULTIPLE: "data-toggle-multiple",
  // common
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_HIDDEN: "aria-hidden",
}

export const Messages = {
  NO_ACCORDION_ID_ERROR:
    "Could not initialize accordion; you must include a value for the 'data-accordion' attribute.",
  NO_ACCORDION_ERROR: id => `Could not find element matching [data-accordion='${id}']`,
  TRIGGERS_TO_COLLAPSIBLES_LENGTH_WARNING: (id, triggersLength, collapsiblesLength) =>
    `Your accordion with id '${id}' has ${triggersLength} triggers and ${collapsiblesLength} collapsibles; make sure there is a trigger for each collapsible!`,
}
