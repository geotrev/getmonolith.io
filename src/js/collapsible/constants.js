export const Selectors = {
  // unique
  DATA_COLLAPSIBLE: "data-collapsible",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_CONTROLS: "aria-controls",
  ARIA_HIDDEN: "aria-hidden",
  ARIA_LABELLEDBY: "aria-labelledby",
  TABINDEX: "tabindex",
  // classes
  TRIGGER_READY_CLASS: "collapsible-trigger-ready",
  CONTENT_READY_CLASS: "collapsible-content-ready",
}

export const CssProperties = {
  HEIGHT: "height",
  VISIBILITY: "visibility",
}

export const CssValues = {
  AUTO: "auto",
  HIDDEN: "hidden",
  VISIBLE: "visible",
}

export const Events = {
  CLICK: "click",
  TRANSITIONEND: "transitionend",
}

export const Messages = {
  NO_COLLAPSIBLE_ID_ERROR:
    "Could not initialize collapsible; you must include a value for the 'data-collapsible' attribute.",
  NO_TRIGGER_ERROR: id =>
    `Could not find collapsible trigger with [data-target='${id}']; you can't have a collapsible without a trigger.`,
  NO_TRIGGER_ID_ERROR: id => `Could not find id on collapsible trigger with [data-target='${id}'].`,
  NO_CONTENT_ERROR: id =>
    `Could not find collapsible content with id '${id}'; you can't have a collapsible without content.`,
}
