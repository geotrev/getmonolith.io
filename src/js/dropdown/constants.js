export const KeyCodes = {
  TAB: 9,
  SHIFT: 16,
  ESCAPE: 27,
  ARROW_UP: 38,
  ARROW_DOWN: 40,
}

export const Selectors = {
  // unique
  DATA_DROPDOWN: "data-dropdown",
  DROPDOWN_MENU_CLASS: "dropdown-menu",
  // common
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  // accessibility
  TABINDEX: "tabindex",
  ARIA_HASPOPUP: "aria-haspopup",
  ARIA_CONTROLS: "aria-controls",
  ARIA_LABELLEDBY: "aria-labelledby",
  ARIA_EXPANDED: "aria-expanded",
  // classes
  OVERLAY_OPEN: "overlay-open",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  BLUR: "blur",
}

export const Messages = {
  NO_DROPDOWN_ID_ERROR:
    "Could not setup dropdown. Make sure it has a valid [data-dropdown] attribute with a unique id as its value.",
  NO_MENU_ERROR: attr => `Could not find menu associated with ${attr}.`,
  NO_DROPDOWN_ITEMS_ERROR: attr => `Could not find any list items associated with ${attr}.`,
  NO_DROPDOWN_ACTIONS_ERROR: attr =>
    `Could not find any button or anchor elements associated with ${attr}.`,
  NO_PARENT_ERROR: "Could not find dropdown button's [data-parent] attribute.",
}
