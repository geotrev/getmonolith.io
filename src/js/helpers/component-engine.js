import { queryAll } from "./dom"
import { isBrowserEnv } from "./is-browser-env"

const isString = value => typeof value === "string"
const isFunction = value => typeof value === "function"

export const ComponentEngine = {
  /**
   * Initialize an Undernet component globally or by id.
   * @param {Object} metadata
   * @property {string} id
   * @property {string} attribute
   * @property {Object} thisArg
   */
  start: (metadata = {}) => {
    if (!isBrowserEnv) return

    const { id, attribute, thisArg } = metadata

    if (id && isString(id)) {
      const instance = document.querySelector(`[${attribute}='${id}']`)
      if (!instance) return

      const validComponent = [instance].filter(thisArg._validate)[0]
      if (!validComponent) return

      thisArg._components.push(validComponent)
    } else if (!id && !thisArg._components.length) {
      const instances = queryAll(`[${attribute}]`)
      if (!instances.length) return

      const validComponents = instances.filter(thisArg._validate)
      thisArg._components = thisArg._components.concat(validComponents)
    } else {
      // attempted to .start() when .stop() wasn't run,
      // OR tried to instantiate a component that's already active.
    }
  },
  /**
   * Teardown an Undernet component globally or by id.
   * @param {Object} metadata
   * @property {string} id
   * @property {string} attribute
   * @property {Object} thisArg
   * @property {string} activeNodeKey
   * @property {*=} cancelActiveFn
   */
  stop: (metadata = {}) => {
    if (!isBrowserEnv) return

    const { id, attribute, thisArg, activeNodeKey, cancelActiveFn } = metadata

    if (id && isString(id)) {
      let targetIndex
      const instance = thisArg._components.filter((activeInstance, index) => {
        if (activeInstance.getAttribute(attribute) !== id) return false
        targetIndex = index
        return true
      })[0]

      if (!instance) return

      if (
        thisArg[activeNodeKey] &&
        instance === thisArg[activeNodeKey] &&
        isFunction(cancelActiveFn)
      )
        thisArg[cancelActiveFn]()

      thisArg._teardown(instance)
      thisArg._components.splice(targetIndex, 1)
    } else if (!id && thisArg._components.length) {
      if (thisArg[activeNodeKey] && isFunction(cancelActiveFn)) thisArg[cancelActiveFn]()

      thisArg._components.forEach(thisArg._teardown)
      thisArg._components = []
    }
  },
}
