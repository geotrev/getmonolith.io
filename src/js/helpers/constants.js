export const KeyCodes = {
  SHIFT: 16,
  TAB: 9,
  ARROW_UP: 38,
  ARROW_DOWN: 40,
}

export const Selectors = {
  FOCUSABLE_TAGS: ["a", "button", "input", "object", "select", "textarea", "[tabindex]"],
  KEYBOARD_CLASS: "using-keyboard",
  NOT_VISUALLY_HIDDEN_CLASS: ":not(.is-visually-hidden)",
  TABINDEX: "tabindex",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  BLUR: "blur",
}

export const Messages = {
  NO_SELECTOR_STRING_OR_CHILDREN_ERROR:
    "createFocusTrap must be given one or both of: first parameter (as selector string)" +
    " and/or options.children (array of elements).",
  OPTION_MATCHERS_DATA_TYPE_ERROR:
    "Invalid data type given to options.matchers for createFocusTrap. Expected: Array.",
  INCORRECT_MATCHER_TYPE_ERROR: type =>
    `Invalid matcher given to options.matchers for createFocusTrap. Expected: String. Recieved: ${type}.`,
  NO_MATCHER_LENGTH_ERROR:
    "Invalid value given to options.matchers for createFocusTrap; value must be an array with at least one selector string",
}
