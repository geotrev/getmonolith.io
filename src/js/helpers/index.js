export * from "./component-engine"
export { createFocusRing } from "./create-focus-ring"
export { createFocusTrap } from "./create-focus-trap"
export * from "./dom"
export { focusOnce } from "./focus-once"
export { isBrowserEnv } from "./is-browser-env"
export { isiOSMobile } from "./is-ios-mobile"
export { log } from "./log"
export { throttle } from "./throttle"
