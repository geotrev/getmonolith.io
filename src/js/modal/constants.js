export const KeyCodes = {
  ESCAPE: 27,
}

export const Selectors = {
  // unique
  DATA_MODAL: "data-modal",
  // common
  DATA_TARGET: "data-target",
  DATA_VISIBLE: "data-visible",
  DATA_CLOSE: "data-close",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_HIDDEN: "aria-hidden",
  ARIA_MODAL: "aria-modal",
  ROLE: "role",
  TABINDEX: "tabindex",
  // classes
  NO_SCROLL: "no-scroll",
}

export const CssProperties = {
  PADDING_RIGHT: "paddingRight",
  PADDING_LEFT: "paddingLeft",
  CURSOR: "cursor",
}

export const CssValues = {
  AUTO: "auto",
  POINTER: "pointer",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  RESIZE: "resize",
  BLUR: "blur",
}

export const Messages = {
  NO_TRIGGER_ERROR: id => `Could not find modal trigger with id ${id}.`,
  NO_ID_ERROR:
    "Could not detect an id on your [data-modal] element. Please add a value matching the modal trigger's [data-parent] attribute.",
  NO_MODAL_DIALOG_ERROR: id => `Could not find element with attribute [data-parent='${id}'].`,
}
