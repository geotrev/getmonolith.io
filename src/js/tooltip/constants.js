export const KeyCodes = {
  ESCAPE: 27,
}

export const Selectors = {
  // unique
  DATA_TOOLTIP: "data-tooltip",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ROLE: "role",
  ARIA_DESCRIBEDBY: "aria-describedby",
  // classes
  DROP_INLINE_START_CLASS: "is-drop-inline-start",
  DROP_INLINE_END_CLASS: "is-drop-inline-end",
  OVERLAY_OPEN: "overlay-open",
}

export const CssProperties = {
  HEIGHT: "height",
  WIDTH: "width",
  TOP: "top",
  LEFT: "left",
}

export const Events = {
  CLICK: "click",
  MOUSEOVER: "mouseover",
  MOUSEOUT: "mouseout",
  FOCUS: "focus",
  BLUR: "blur",
  KEYDOWN: "keydown",
}

export const Messages = {
  NO_ID_ERROR: "Could not find tooltip id.",
  NO_TRIGGER_ERROR: id => `Could not find a tooltip trigger with attribute [data-target='${id}'].`,
  NO_TOOLTIP_ERROR: id => `Could not find a tooltip with id '${id}'.`,
}
