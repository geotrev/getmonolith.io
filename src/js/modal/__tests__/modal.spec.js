import Undernet from "../../"
import { find, renderDOM, simulateKeyboardEvent } from "../../helpers/test"
import { KeyCodes, Messages } from "../constants"

const dom = `
  <button data-target="modal-id">Open modal</button>

  <div class="modal-overlay" data-modal="modal-id">
    <div class="modal-dialog" data-parent="modal-id" aria-labelledby="header-id">
      <header>
        <h2 class="h6 has-no-m-block-start" id="header-id">
          Modal Header
        </h2>
        <a data-close href="#">
          <span aria-hidden="true">&times;</span>
        </a>
      </header>
      <section>
        <p>Some modal content here</p>
      </section>
      <footer>
        <a class="button" data-close href="#">
          Cancel
        </a>
        <a class="primary button" href="#">
          OK
        </a>
      </footer>
    </div>
  </div>
`

console.error = jest.fn()

const activeElement = () => document.activeElement

const mockTransitionEnd = () => {
  const modal = find("[data-modal='modal-id']")
  const spy = jest.spyOn(modal, "addEventListener")

  spy.mockImplementation((event, fn) => (event === "transitionend" ? fn() : fn))

  return spy
}

describe("Modal", () => {
  describe("API start", () => {
    it("sets attributes", () => {
      // Given
      const wrapper = renderDOM(dom)
      // When
      Undernet.Modals.start()
      // Then
      expect(wrapper).toMatchSnapshot()
      Undernet.Modals.stop()
    })
  })

  describe("API stop + #handleClick", () => {
    it("does not open modal", () => {
      // Given
      const wrapper = renderDOM(dom)
      const spy = mockTransitionEnd()

      const trigger = find("[data-target='modal-id']")
      // When
      Undernet.Modals.start()
      Undernet.Modals.stop()
      trigger.click()
      // Then
      expect(wrapper).toMatchSnapshot()
      spy.mockRestore()
      Undernet.Modals.stop()
    })
  })

  describe("#handleClick", () => {
    let wrapper, spy

    beforeEach(() => {
      wrapper = renderDOM(dom)
      spy = mockTransitionEnd()

      const trigger = find("[data-target='modal-id']")

      Undernet.Modals.start()
      trigger.click()
    })

    afterEach(() => {
      spy.mockRestore()
      Undernet.Modals.stop()
    })

    it("opens modal", () => {
      expect(wrapper).toMatchSnapshot()
    })

    it("sets focus to modal dialog", () => {
      const modalDialog = find("[data-parent]")
      expect(activeElement()).toEqual(modalDialog)
    })
  })

  describe("#handleClose", () => {
    let wrapper, trigger, spy

    beforeEach(() => {
      wrapper = renderDOM(dom)

      spy = mockTransitionEnd()
      trigger = find("[data-target='modal-id']")
      const closeButton = find("header [data-close]")

      Undernet.Modals.start()
      trigger.click()
      closeButton.click()
    })

    afterEach(() => {
      spy.mockRestore()
      Undernet.Modals.stop()
    })

    it("closes modal", () => {
      expect(wrapper).toMatchSnapshot()
    })

    it("sets focus to modal trigger", () => {
      expect(activeElement()).toEqual(trigger)
    })
  })

  describe("#handleOverlayClick", () => {
    it("closes modal", () => {
      // Given
      const wrapper = renderDOM(dom)
      const spy = mockTransitionEnd()
      const trigger = find("[data-target='modal-id']")
      // When
      Undernet.Modals.start()
      trigger.click()
      find("[data-modal='modal-id']").click()
      // Then
      expect(wrapper).toMatchSnapshot()
      spy.mockRestore()
      Undernet.Modals.stop()
    })
  })

  describe("#handleEscapeKeyPress", () => {
    it("closes modal", () => {
      // Given
      const wrapper = renderDOM(dom)
      const spy = mockTransitionEnd()
      const trigger = find("[data-target='modal-id']")
      // When
      Undernet.Modals.start()
      trigger.click()
      simulateKeyboardEvent(KeyCodes.ESCAPE)
      // Then
      expect(wrapper).toMatchSnapshot()
      spy.mockRestore()
      Undernet.Modals.stop()
    })
  })

  describe("#handleReturnFocus", () => {
    it("sets focus back to modal trigger", () => {
      // Given
      renderDOM(dom)

      const spy = mockTransitionEnd()
      const trigger = find("[data-target='modal-id']")
      const closeButton = find("header [data-close]")
      // When
      Undernet.Modals.start()
      trigger.click()
      closeButton.click()
      // Then
      expect(activeElement()).toEqual(trigger)
      spy.mockRestore()
      Undernet.Modals.stop()
    })
  })

  describe("#handleScrollRestore", () => {
    let spy

    beforeEach(() => {
      renderDOM(dom)

      spy = mockTransitionEnd()
      const trigger = find("[data-target='modal-id']")
      const closeButton = find("header [data-close]")

      Undernet.Modals.start()
      trigger.click()
      closeButton.click()
    })

    afterEach(() => {
      spy.mockRestore()
      Undernet.Modals.stop()
    })

    it("removes 'no-scroll' class from <body>", () => {
      expect(find("body").className).toBe("")
    })

    it("removes 'no-scroll' class from <html>", () => {
      expect(find("html").className).toBe("")
    })
  })

  describe("#handleScrollStop", () => {
    let spy

    beforeEach(() => {
      renderDOM(dom)

      spy = mockTransitionEnd()
      const trigger = find("[data-target='modal-id']")

      Undernet.Modals.start()
      trigger.click()
    })

    afterEach(() => {
      spy.mockRestore()
      Undernet.Modals.stop()
    })

    it("sets 'no-scroll' class to <body>", () => {
      expect(find("body").className).toBe("no-scroll")
    })

    it("sets 'no-scroll' class to <html>", () => {
      expect(find("html").className).toBe("no-scroll")
    })
  })
})

const errorDom = (target, modal, parent) => `
  <button data-target="${target}">Open modal</button>

  <div className="modal-overlay" data-modal="${modal}">
    <div className="modal-dialog" data-parent="${parent}" aria-labelledby="header-id">
      <header>
        <h2 className="h6 has-no-m-block-start" id="header-id">
          Modal Header
        </h2>
        <a data-close href="#">
          <span aria-hidden="true">&times;</span>
        </a>
      </header>
      <section>
        <p>Some modal content here</p>
      </section>
      <footer>
        <a className="button" data-close href="#">
          Cancel
        </a>
        <a className="primary button" href="#">
          OK
        </a>
      </footer>
    </div>
  </div>
`

describe("Modal Warnings", () => {
  const MODAL_ID = "modal-id"

  it("prints console error if modal trigger isn't found", () => {
    // Given
    renderDOM(errorDom("", MODAL_ID, MODAL_ID))
    // When
    Undernet.Modals.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_TRIGGER_ERROR(MODAL_ID))
  })

  it("prints console error if modal id isn't found", () => {
    // Given
    renderDOM(errorDom(MODAL_ID, "", MODAL_ID))
    // When
    Undernet.Modals.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_ID_ERROR)
  })

  it("prints console error if [data-parent] attribute does not match its parent [data-modal]", () => {
    // Given
    renderDOM(errorDom(MODAL_ID, MODAL_ID, ""))
    // When
    Undernet.Modals.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_MODAL_DIALOG_ERROR(MODAL_ID))
  })
})
