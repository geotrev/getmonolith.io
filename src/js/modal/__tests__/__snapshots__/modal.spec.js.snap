// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal #handleClick opens modal 1`] = `
<body
  class="no-scroll"
  style="padding-right: 1024px;"
>
  
  
  <button
    data-target="modal-id"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="false"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="true"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
      tabindex="-1"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="0"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="0"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="0"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;

exports[`Modal #handleClose closes modal 1`] = `
<body
  class=""
  style="padding-right: 1024px;"
>
  
  
  <button
    data-target="modal-id"
    tabindex="-1"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="true"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="false"
    style="padding-left: 1024px;"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="-1"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;

exports[`Modal #handleEscapeKeyPress closes modal 1`] = `
<body
  class=""
  style="padding-right: 1024px;"
>
  
  
  <button
    data-target="modal-id"
    tabindex="-1"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="true"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="false"
    style="padding-left: 1024px;"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="-1"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;

exports[`Modal #handleOverlayClick closes modal 1`] = `
<body
  class=""
  style="padding-right: 1024px;"
>
  
  
  <button
    data-target="modal-id"
    tabindex="-1"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="true"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="false"
    style="padding-left: 1024px;"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="-1"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;

exports[`Modal API start sets attributes 1`] = `
<body>
  
  
  <button
    data-target="modal-id"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="true"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="false"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="-1"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;

exports[`Modal API stop + #handleClick does not open modal 1`] = `
<body>
  
  
  <button
    data-target="modal-id"
  >
    Open modal
  </button>
  

  
  <div
    aria-hidden="true"
    classname="modal-overlay"
    data-modal="modal-id"
    data-visible="false"
  >
    
    
    <div
      aria-labelledby="header-id"
      aria-modal="true"
      classname="modal-dialog"
      data-parent="modal-id"
      role="dialog"
    >
      
      
      <header>
        
        
        <h2
          classname="h6 has-no-m-block-start"
          id="header-id"
        >
          
          Modal Header
        
        </h2>
        
        
        <a
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          
          <span
            aria-hidden="true"
          >
            ×
          </span>
          
        
        </a>
        
      
      </header>
      
      
      <section>
        
        
        <p>
          Some modal content here
        </p>
        
      
      </section>
      
      
      <footer>
        
        
        <a
          classname="button"
          data-close=""
          href="#"
          tabindex="-1"
        >
          
          Cancel
        
        </a>
        
        
        <a
          classname="primary button"
          href="#"
          tabindex="-1"
        >
          
          OK
        
        </a>
        
      
      </footer>
      
    
    </div>
    
  
  </div>
  

</body>
`;
